src/main/java/duke/TaskArrayList.java

```diff
@@ -0,0 +1,106 @@
+package duke;
+
+import duke.exceptions.DukeException;
+import duke.tasks.Task;
+
+import java.util.ArrayList;
+
+/**
+ * Array List of Task objects.
+ */
+public class TaskArrayList extends ArrayList<Task> {
+    public final static String DELETE_USAGE_TEXT = "Usage: delete <integer task number>";
+    public final static String DONE_USAGE_TEXT = "Usage: done <integer task number>";
+    public final static String FIND_USAGE_TEXT = "Usage: find <search String>";
+
+    public TaskArrayList() {
+        super();
+    }
+
+    /**
+     * Adds a task to the list.
+     *
+     * @param task Task object to add.
+     * @return addition message String.
+     */
+    public String addTask(Task task) {
+        this.add(task);
+        return task.addMsg() + "\n" + this.announceNewLength();
+    }
+
+    /**
+     * Marks the given task as done.
+     *
+     * @param taskNo task number to mark as done.
+     * @return removal message String.
+     * @throws DukeException when invalid task number provided.
+     */
+    public String deleteTask(int taskNo) throws DukeException {
+        if (taskNo > this.size()) {
+            throw new DukeException(String.format("task %d not found", taskNo));
+        }
+        Task toDel = this.get(taskNo - 1);
+        this.remove(taskNo - 1);
+        return "removed : " + toDel.toString() + "\n" + this.announceNewLength();
+    }
+
+    /**
+     * Generates the message to print when tasklist length changes.
+     *
+     * @return You now have X task(s) in the list.
+     */
+    private String announceNewLength() {
+        return String.format("Now you have %d %s in the list.", this.size(), this.size() == 1 ? "task" : "tasks");
+    }
+
+    /**
+     * Enumerates members for printing
+     *
+     * @return String[] of "X. taskname".
```

Comment:
I think to avoid misunderstanding, perhaps instead of using the term "String[]" to describe the return type, you could say something like"A String showing all the contents of the tasklist." Feel free to use or not use my suggestion though!